src/utils/tokens.ts

```diff
@@ -411,9 +418,9 @@ const tokens = {
     const withdrawAmount = this.getWithdrawAmount(amount, depositPercent);
 
     // get inputs that amount to requested melt amount
-    let foundAmount = 0;
-    for await (const utxo of storage.selectUtxos({token, target_amount: amount})) {
-      foundAmount += utxo.value;
+    const selectedUtxos = await utxoSelection(storage, token, amount);
+    const foundAmount = selectedUtxos.amount;
+    for (const utxo of selectedUtxos.utxos) {
```

Comment:
Maybe we should use the authorities from the utxo instead of `authorities: 0`, if there was an error selecting utxos, this would hide if we chose an authority utxo.

In the same vein, this is actually a conversion from `IUtxo` to `IDataInput` which have very just a few properties of difference, maybe we should have it as an util since this conversion should be very common.

Reply:
Done: https://github.com/HathorNetwork/hathor-wallet-lib/pull/569/commits/733cbd75571c0d7c2d2d3c08f529bafda956fee2